apps/gitness/src/pages-v2/repo/repo-commits.tsx

```diff
@@ -0,0 +1,80 @@
+
+import { useFindRepositoryQuery, useListBranchesQuery, useListCommitsQuery } from '@harnessio/code-service-client'
+import { BranchSelectorListItem, RepoCommitsView } from '@harnessio/ui/views'
+
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { useTranslationStore } from '../../i18n/stores/i18n-store'
+import { PathParams } from '../../RouteDefinitions'
+import { PageResponseHeader } from '../../types'
+import { normalizeGitRef } from '../../utils/git-utils'
+
+const sortOptions = [{ name: 'Sort option 1' }, { name: 'Sort option 2' }, { name: 'Sort option 3' }]
+
+export default function RepoCommitsPage() {
+  const repoRef = useGetRepoRef()
+  const { spaceId, repoId } = useParams<PathParams>()
+
+  const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
+  const [sort, setSort] = useQueryState('sort')
+  const [query, setQuery] = useQueryState('query')
+
+  const { data: { body: repository } = {} } = useFindRepositoryQuery({ repo_ref: repoRef })
+  const { data: { body: branches } = {}, isFetching: isFetchingBranches } = useListBranchesQuery({
+    repo_ref: repoRef,
+    queryParams: { page }
+  })
+
+  const [selectedBranch, setSelectedBranch] = useState<BranchSelectorListItem>({ name: '', sha: '' })
+
+  const { data: { body: commitData, headers } = {}, isFetching: isFetchingCommits } = useListCommitsQuery({
+    repo_ref: repoRef,
+    queryParams: { page, git_ref: normalizeGitRef(selectedBranch?.name), include_stats: true }
+  })
+
+  const xNextPage = parseInt(headers?.get(PageResponseHeader.xNextPage) || '')
+  const xPrevPage = parseInt(headers?.get(PageResponseHeader.xPrevPage) || '')
+
+  // 🚨 API not supporting sort, so waiting for API changes
+  // const { sort } = useCommonFilter()
+
+  useEffect(() => {
+    if (repository) {
+      const defaultBranchSha = branches?.find(branch => branch.name === repository?.default_branch)?.sha || ''
+      setSelectedBranch({ name: repository?.default_branch || '', sha: defaultBranchSha })
+    }
+  }, [repository])
+
+  const selectBranch = (branch: BranchSelectorListItem) => {
+    setSelectedBranch(branch)
+  }
+
+  return (
+    <RepoCommitsView
+      branches={branches?.map(branch => {
+        return { name: branch.name, sha: branch.sha }
+      })}
+      commitsList={commitData?.commits}
+      isFetchingBranches={isFetchingBranches}
+      isFetchingCommits={isFetchingCommits}
+      page={page}
+      selectBranch={selectBranch}
+      selectedBranch={selectedBranch}
+      setPage={(page: number) => setPage(page)}
+      xNextPage={xNextPage}
+      xPrevPage={xPrevPage}
+      sortOptions={sortOptions}
+      sort={sort}
+      setSort={(sort: string) => setSort(sort)}
+      query={query}
+      setQuery={(query: string) => setQuery(query)}
+      repoId={repoId || ''}
+      spaceId={spaceId || ''}
+      tagList={[]}
+      useTranslationStore={useTranslationStore}
+    />
```

Comment:
I am assuming this will be moved into a store as a follow up pr?

Reply:
Right